src/Polly.Net35/CircuitBreaker/SuccessRatioCircuitBreakerState.cs

```diff
@@ -47,6 +53,7 @@ public void Reset()
             using (TimedLock.Lock(_lock))
             {
                 _successCount += 1;

```

Comment:
So the current counters are data points of what the ratio was in the past until `Decay` gets called. Then the counts become current. Whenever the counts are access or modified we need to bring them up to what they ought to be in the present by decaying them.

Lets say we have a half life of 1 minute.
1. A request happens and I increment a counter so our success/fail is 1/0
2. 1 minute passes. Another request happens. At this point the values are still 1/0 but 1 minute has gone by and we want the values to actually be 0.5/0. We call `Decay` to get us there. The final values after the increments should be 1.5/0

One more example - 
half life = 1 minute
success count = 30
failure count = 10

1 minute later the values should be
success count = 15
failure count = 5

2 minutes later the values should be
success count = 7.5
failure count = 2.5